manager/integration/tests/test_ha.py

```diff
@@ -2426,3 +2426,94 @@ def test_replica_failure_during_attaching(settings_reset, client, core_api, volu
     del update_disks["extra-disk"]
     node.diskUpdate(disks=update_disks)
     common.wait_for_disk_update(client, node.name, 1)
+
+
+@pytest.mark.skip(reason="TODO") # NOQA
+def test_engine_image_missing_on_some_nodes():
+    """
+    This e2e test follows the manual test steps at:
+    https://github.com/longhorn/longhorn/issues/2081#issuecomment-783747541
+
+    Steps:
+
+    Preparation:
+    1. Set up a backup store
+    2. Let's name the 3 nodes: node-1, node-2, node-3
+
+    Case 1: Test volume operations when engine image DaemonSet is miss
+    scheduled
+    1. Create a volume, vol-1, of 3 replicas
+    2. Create another volume, vol-2, of 3 replicas
+    3. Taint node-1 with the taint: key=value:NoSchedule
+    4. Verify that all functions (attach, snapshot, backup,
+       expand, detach) are working ok for vol-1
+
+    Case 2: Test volume operations when engine image DaemonSet is not fully
+    deployed
+    1. Attach vol-1 to node-1. Change the number of replicas of vol-1
+       to 2. Delete the replica on node-1
+    2. Delete the pod on node-1 of the engine image DaemonSet.
+       Or delete the engine image DaemonSet and wait for Longhorn
+       to automatically recreates it.
+    3. Wait for the engine image CR state become deploying
+    4. Verify that functions (snapshot, backup, detach) are working ok
+       for vol-1
+    5. Detach vol-1
+    6. Attach vol-1 to node-1. Verify that Longhorn cannot attach vol-1 to
+       node-1 since there is no engine image on node-1
+    7. Verify that all functions (attach to other nodes, snapshot, backup,
+       expand, detach) are working ok for vol-1
+    8. Verify that vol-2 cannot be attached to any nodes because one of
```

Comment:
Will the volume get stuck in attaching? Or will the `attach` call directly error out?

Reply:
Thank you. Fixed it to specify that we should get the API error